benchalerts/clients.py

```diff
@@ -248,37 +249,32 @@ def get_comparison_to_baseline(
         dict
             A dict where keys are contender run_ids and values are lists of dicts
             containing benchmark comparison information.
+        bool
+            True if all the baseline runs were found on the immediate parent of the
+            contender commit. If False, the contender might be a non-first PR commit, or
+            there could have been commits on the default branch without any logged
+            Conbench runs.
         """
-        contender_info = self.get("/commits/", params={"sha": contender_sha})
-        if len(contender_info) != 1:
+        comparisons = {}
+        baseline_is_parent = True
+        contender_runs = self.get("/runs/", params={"sha": contender_sha})
+        if not contender_runs:
             fatal_and_log(
-                f"Found {len(contender_info)} commits in conbench that match the "
-                f"contender SHA '{contender_sha}'."
+                f"Contender commit '{contender_sha}' doesn't have any runs in conbench."
             )
 
-        baseline_sha = contender_info[0].get("parent_sha")
-        if not baseline_sha:
-            fatal_and_log(
-                f"Found the contender commit ({contender_sha}) but it doesn't have a "
-                "baseline commit in conbench."
-            )
+        log.info(f"Getting comparisons from {len(contender_runs)} runs")
+        for run in contender_runs:
+            contender_info = self.get(f"/runs/{run['id']}/")
+            baseline_path = contender_info["links"]["baseline"].split("/api")[-1]
+            baseline_info = self.get(baseline_path)
 
-        commit_compare = self.get(f"/compare/commits/{baseline_sha}...{contender_sha}")
-        all_runs = commit_compare["runs"]
-        if not all_runs:
-            fatal_and_log(
-                f"Contender commit '{contender_sha}' doesn't have any runs in conbench."
-            )
+            if baseline_info["commit"]["sha"] != contender_info["commit"]["parent_sha"]:
+                baseline_is_parent = False
 
-        comparisons = {}
-        log.info(f"Getting comparisons from {len(all_runs)} runs")
-        for run in all_runs:
-            path = (
-                "/compare/runs/"
-                f"{run['baseline']['run_id']}...{run['contender']['run_id']}"
-            )
+            path = f"/compare/runs/{baseline_info['id']}...{contender_info['id']}"
             params = {"threshold_z": z_score_threshold} if z_score_threshold else None
             comparison = self.get(path, params=params)
-            comparisons[run["contender"]["run_id"]] = comparison
+            comparisons[contender_info["id"]] = comparison
 
-        return comparisons
+        return comparisons, baseline_is_parent
```

Comment:
nit (please ignore unless it sparks an idea): I don't love the multiple return only used in testing, but short of a pretty major rearrangement I don't see a trivial way around it. If this were an internal method that started with `_` I think I'd care less

Reply:
yeah, it's a little messy right now. The major thing that this bool (and the data structure you mentioned too) will power is when we set up the Checks API, we can have a nice lil markdown report clickable from PRs. So we can iterate over these messy outputs and format them into links, warnings, etc for that report.

I've been thinking about leading `_`s but really most of the stuff in this package is internal and probably won't even be looked at by most people. So the slight inconvenience of dealing with `_`s everywhere doesn't even seem worth it right now